pyuvdata/uvdata/uvdata.py

```diff
@@ -6070,6 +6069,90 @@ def set_uvws_from_antenna_positions(self, update_vis=True):
         self.uvw_array = new_uvw
         return
 
+    def update_antenna_positions(
+        self, new_positions=None, delta_antpos=False, update_vis=True
+    ):
+        """
+        Update antenna positions and associated (meta)data.
+
+        This method will update the antenna positions for a UVData object, and
+        correspondingly will adjust the uvw-coordinates for each of the baselines, as
+        well as the phases of the visibilities (if requested).
+
+        Parameters
+        ----------
+        new_positions : dict
+            Dictionary containing new antenna positions, where the key is the antenna
+            number (which should match to an entry in UVData.antenna_numbers), and the
+            value is a 3-element array corresponding to the ECEF/MCMF position relative
+            to the array center.
+        delta_antpos : bool
+            When set to True, uvws are updated be calculating the difference between
+            the old and new antenna positions. Thiis option should be used with care,
+            and should only be used when warrented (e.g., antenna positions are stored
+            with higher positions than the baselines). Default is False.
```

Comment:
Bah, I really mucked up this docstring. But your attempt to interpret my gibberish is spot on -- sometimes the baseline uvws aren't recorded with enough precision (32-bit floats for MIR), and so it can be useful to use the difference in antenna positions (stored as 64-bit floats, or technically as long strings of ascii text with 15 digits -- it's a weird format) in lieu of said baselines. Fixed the docstring in ac42d92d3a50e55de06e2aadf34d30236cccf9df.